src/components/SchedulePanel/index.js

```diff
@@ -0,0 +1,361 @@
+    this.handleEditDescriptionOpen = this.handleEditDescriptionOpen.bind(this);
+    this.handleEditDescriptionClose = this.handleEditDescriptionClose.bind(this);
+    this.handleEditDescriptionButton = this.handleEditDescriptionButton.bind(this);
+    this.handleDoubleClick = this.handleDoubleClick.bind(this);
+    this.handleRemoveSubmit = this.handleRemoveSubmit.bind(this);
+  }
+
+  componentWillMount() {
+    this.loadData(this.props.profile);
+    this.setState({ isExpanded: true });
+  }
+
+  loadData = async searchedUser => {
+    try {
+      const scheduleControlPromise = schedulecontrol.getScheduleControl(searchedUser.AD_Username);
+      const scheduleControlInfo = await scheduleControlPromise;
+      this.scheduleControlInfo = scheduleControlInfo;
+    } catch (e) {
+      this.setState({ loading: false });
+    }
+    if (this.scheduleControlInfo) {
+      this.setState({ isSchedulePrivate: this.scheduleControlInfo.IsSchedulePrivate });
+      this.setState({ 
+        description: this.scheduleControlInfo.Description ? 
+        this.scheduleControlInfo.Description.replace(new RegExp("SlSh", 'g'), '/').replace(new RegExp("CoLn", 'g'), ':')
+      .replace(new RegExp("dOT", 'g'), '.') : ''});
+      this.setState({ modifiedTimeStamp: this.scheduleControlInfo.ModifiedTimeStamp });
+    }
+    console.log('Schedule Control : ', this.scheduleControlInfo);
+    this.setState({ loading: false });
+  };
+
+  handleOfficeHoursOpen = slotInfo => {
+    this.setState({ officeHoursOpen: true });
+    if (slotInfo){
+      let startTime = slotInfo.start.toTimeString().split(":");
+      let endTime = slotInfo.end.toTimeString().split(":");
+      this.setState({start:startTime[0] + ":" + startTime[1],
+        end:endTime[0] + ":" + endTime[1], resourceId: slotInfo.resourceId,
+      });
+    }
+    this.setState({ isDoubleClick: false });
+  };
+
+  handleOfficeHoursClose = () => {
+    this.setState({
+      officeHoursOpen: false,
+      isDoubleClick: false,
+      selectedEvent: null,
+    });
+  };
+
+  handleRemoveOfficeHoursOpen = () => {
+    this.setState({ removeOfficeHoursOpen: true });
+  };
+
+  handleRemoveOfficeHoursClose = () => {
+    this.setState({ removeOfficeHoursOpen: false });
+  };
+
+  handleRemoveButton = event => {
+    if (event.id > 1000) {
+      this.setState({ disabled: false, selectedEvent: event }, () => {
+        console.log(this.state.selectedEvent)});
+    } else {
+      this.setState({ disabled: true});
+    }
+  };
+
+  handleEditDescriptionOpen = () => {
+    this.setState({ editDescriptionOpen: true });
+  };
+
+  handleEditDescriptionClose = () => {
+    this.setState({ editDescriptionOpen: false });
+  };
+
+  handleEditDescriptionButton = () => {
+    this.setState({ disabled: false });
+  };
+
+  handleDescriptionSubmit = async descValue => {
+    await schedulecontrol.setScheduleDescription(descValue);
+    window.location.reload(); // refresh to show the change
```

Comment:
Is there no better way to display the change? Could you just run loadData again?

Reply:
Change made